crates/virtual-dom-rs/src/diff.rs

```diff
@@ -9,6 +9,18 @@ pub fn diff<'a>(old: &'a VirtualNode, new: &'a VirtualNode) -> Vec<Patch<'a>> {
     diff_recursive(&old, &new, &mut 0)
 }
 
+
+
+fn clean_children<'a, 'b>(
```

Comment:
Thanks for writing this! A couple things that come to mind for me:

1. In terms of vertical hierarchy of the file it feels like this should go below `diff_recursive` since `diff_recursive` uses this

2. `clean_children` doesn't feel like it entirely explains what's going on in here. Maybe `increment_node_idx_for_children`.. ?

3. Could we add a minimal new test case to the bottom of the file that confirms that we're generating the correct patch when we're removing children? Here's one example test 

https://github.com/chinedufn/percy/blob/master/crates/virtual-dom-rs/src/diff.rs#L109-L123

Please let me know if I can clarify any further, especially about point number 3!!

Reply:
All done.